### ISSUE TITLE
EOS Permissions 

### ISSUE THREAD

**Issue by jinjin08021** [EOSIO/eos#6107]
I created admin account with associated accounts like the follow: 

`cleos get account admin`
`     owner     1:    1 alice@owner, 1 carol@owner, 1 bob@owner, 
        active     1:    1 alice@active, 1 carol@active, 1 bob@active, 
`

In my contract, I have this assert statement. 
`eosio_assert(has_auth(N(admin)),"has to have valid authority");`

When alice calls this contract with its own active permission, it fails this assert statement. 

I thought when admin has the following accounts with the threshold of 1, any of the account can act on behalf of admin. Am I missing something here?

**Comment by andriantolie** [comment]
`has_auth` and `require_auth` is checking the permissions inside the action object. When you call the contract using alice's active permission, you are actually appending alice's active permission to the action instead of admin's active permission. This explains the situation that you have above.

> I thought when admin has the following accounts with the threshold of 1, any of the account can act on behalf of admin.
Your above comment overall is correct. However, to be more precise, what it means is that you can "sign the transaction with Alice's key" but you are still **required** to append admin's active.

To summarize it in a practical way, "-p admin" is needed in your transaction, but as long as you have Alice's private key in your wallet, you won't have any problem signing (and acting) as behalf of admin.
